lightwood/mixer/helpers/residual_net.py

```diff
@@ -38,15 +38,22 @@ def __init__(self,
                  input_size: int = None,
                  output_size: int = None,
                  shape: List[int] = None,
-                 max_params: int = int(3e5)) -> None:
+                 max_params: int = int(3e5),
+                 device: str = '') -> None:
         super(ResidualNet, self).__init__()
         self.net = torch.nn.Sequential(
             *
             ([ResidualModule(input_size) for _ in range(1)] +
              [nn.Linear(input_size, max([input_size * 2, output_size * 2, 400])),
               nn.Linear(max([input_size * 2, output_size * 2, 400]),
                         output_size)]))
-        self.to(*get_devices())
+
+        self.to(get_device_from_name(device))
```

Comment:
Yeah, this will fail. Notice the `self.to()` definition: it takes the `device` but also the amount of `available_devices`.

I think one option here would be to turn `available_devices` in `self.to` into an `Optional[int] = 1` argument.

Reply:
Maybe one could have a function that merges the behaviour of `get_devices` and `get_device_from_name`, for example adding an optional `device_str=''` to the former and returns the number of available devices in the cuda case.